servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -314,18 +321,118 @@ private static boolean executeOne(Queue<RunnableWrapper> tasks, AtomicInteger ta
         return false;
     }
 
-    // Wraps Runnables to ensure that object-equality (and hashcode) is used for removal from Lists.
-    // Also ensures a unique object each time, so the same Runnable can be executed multiple times.
+    /**
+     *  Wraps Runnables to ensure that object-equality (and hashcode) is used for removal from Lists.
+     *  Also ensures a unique object each time, so the same Runnable can be executed multiple times.
+     *  Sets the thread name to {@code TestExecutor-#} while running the task so that capturing the thread name makes
+     *  sense and during debugging the execution context is more obvious.
+     *  Adversarially set the {@link AsyncContextMap} to a hostile instance to ensure that any use of
+     *  {@link AsyncContextMap} within the context of the Runnable includes appropriate setting/restoring of the
+     *  context.
+     */
     private static final class RunnableWrapper implements Runnable {
+        private final String threadName;
         private final Runnable delegate;
 
-        private RunnableWrapper(final Runnable delegate) {
+        private RunnableWrapper(final String threadName, final Runnable delegate) {
+            this.threadName = threadName;
             this.delegate = delegate;
         }
 
         @Override
         public void run() {
-            delegate.run();
+            Thread current = Thread.currentThread();
+            String oldName = current.getName();
+            current.setName(threadName);
+            AsyncContextMap tlPrev = contextThreadLocal.get();
+            contextThreadLocal.set(InvalidAsyncContextMap.INSTANCE);
```

Comment:
Can we just keep the thread name change here, and wrap this `RunnableWrapper` with `ContextPreservingRunnable`?

If not, consider changing `contextThreadLocal` only if the thread is not an instance of `AsyncContextMapHolder`, to be consistent with `ContextPreservingRunnable`.

Reply:
Yes, switching to providing `InvalidAsyncContext` instance rather than `null` allowed wrapping with `AsyncContext.provider().wrapRunnable(delegate, InvalidAsyncContextMap.INSTANCE)` and greatly simplifies the implementation to just the thread naming.